File: Algorithms/Sorting/Bubble Sort/bubble-sort.go
/*
 * Bubble Sort
 *
 */

package main

import (
	"fmt"
	"math/rand"
)

func randomArray(n int) []int {
	arr := make([]int, n)
	for i := 0; i < n; i++ {
		arr[i] = rand.Intn(n)
	}
	return arr
}

func bubbleSort() {
	arr := randomArray(10)
	fmt.Println("Bubble Sort")
	fmt.Println("Initial array is: ", arr)

	temp := 0

	for i := 0; i < len(arr); i++ {
		for j := 0; j < len(arr)-1; j++ {
			if arr[j] > arr[j+1] {
				temp = arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = temp
			}
		}
	}

	fmt.Println("Sorted array is: ", arr)
	fmt.Println()
}

func main() {
	bubbleSort()
}

File: Algorithms/Sorting/Bubble Sort/bubble-sort.rb
=begin
    Bubble Sort
=end

def BubbleSort(arr)
    n = arr.length
    j = 0
    i = 0
    while i < n
        while j < n-i-1
            if arr[j] > arr[j+1]
                arr[j], arr[j+1] = arr[j+1], arr[j]
            end
            j += 1
        end
        i += 1
    end
    print arr
end


arr = [3, 1, 5, 8, 11, 10]
BubbleSort(arr)


